ambry-router/src/test/java/com.github.ambry.router/NonBlockingRouterTest.java

```diff
@@ -413,10 +420,12 @@ public void onRequestResponseError(ReplicaId replicaId, ServerErrorCode serverEr
    * @param indexToFail the index representing which response for which failure is to be simulated. For example,
    *                    if index is 0, then the first response will be failed. If the index is -1, no responses will be
    *                    failed.
+   * @param createConnectionUnavailableError if requests are failed, whether the reason should be
+   *                                          NetworkClient.ConnectionUnavailable
    */
   private void testFailureDetectorNotification(OperationHelper opHelper, NetworkClient networkClient,
```

Comment:
I think a more extensible argument would be to pass the `NetworkClientErrorCode` you want.